addons/mail/views/mail_template_views.xml

```diff
@@ -5,44 +5,36 @@
             <field name="name">email.template.form</field>
             <field name="model">mail.template</field>
             <field name="arch" type="xml">
-                <form string="Templates">
+                <form string="Templates" js_class="mail_template_form">
                     <header>
+                        <field name="ref_ir_act_window" invisible="1"/>
                         <field name="template_fs" invisible="1"/>
+                        <field name="current_user_is_template_editor" invisible="1"/>
+                        <button string="Preview"
+                                name="%(mail_template_preview_action)d" type="action"
+                                class="btn-primary" target="new"/>
                         <button string="Reset Template"
                                 name="%(mail_template_reset_action)d" type="action"
                                 groups="mail.group_mail_template_editor"
                                 invisible="not template_fs"/>
+                        <!-- onclick added by js_class for confirmation modal -->
+                        <span class="btn btn-secondary btn-group delete-mail-template"
```

Comment:
Actually I am unsure we really want this, isn't it adding noise as we already have the delete action in the more menu ?

Reply:
We do have it but only accessible to administrator users, since only they can access the form view in a non-wizard. The button is there so that users can delete their own templates when they access the template from a mail.compose.message wizard

Reply:
When playing with it, I have some issues when removing templates in the popup form ... I end up on a void "load template" form that I don't find really user friendly
![template_post_delete](https://github.com/odoo/odoo/assets/7501424/35ff90e5-3623-4375-9676-db139eff4717)

And when closing the popup I end up having issues as template_id is still set on the wizard
![template_post_delete2](https://github.com/odoo/odoo/assets/7501424/346b0e23-fc93-446d-8fc7-bf5a48ff684a)

I am not sure this flow is really a good idea... you are in a wizard, in a transient state, and you want to dynamically remove records. I fear we are going to have several "border line behavior" issues like this one... maybe to discuss with Luc ?